SWIPs/swip-honey_to_money.md

```diff
@@ -7,51 +7,66 @@ Type: Standards track
+This update process should be atomic: either all nodes upgrade or none. If Swarm would update the price by releasing a new version of the source code (as is currently the case), this would be violated as it is not possible to guarantee that all nodes are on the same release. By externalizing the price to an oracle, Swarm completely decouples its core functions from price discovery. By building support for price oracles, it will be possible to coordinate price changes across the network and set the stage for future experimentation with various mechanisms for setting prices, from DAOs, to bonding curves, to fixed prices. 
 
 ## Specification
-* The Swarm source code will reference the address of a `HoneyToMoney` price oracle
-* The price oracle implements the `HoneyToMoney` interface (to-be specified) 
-* The `HoneyToMoney` contract will be initially owned by an M/N multi-sig of Swarm developers and stakeholders.
-* Before sending a payment (denominated in currency), a node will query the `HoneyToMoney` price oracle to get the most up-to-date price conversion. 
-* Upon receiving a payment, nodes will determine the honey amount paid by looking up and dividing by the currently valid price from the oracle. This might cause payment imbalances between nodes, as the oracle might be queried one or several blocks before the transaction was sent. To avoid this situation, we propose that nodes include a payload with the transaction, specifying the time at which the oracle was queried.
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
+* An oracle is reachable at one specific endpoint and this endpoint is hardcoded in the Swarm source-code. 
+* No other factor besides time influences the answer which an oracle gives. That is: if nodes query the oracle at the same time, the oracle gives the same answer. 
+* A price oracle returns a `honeyPrice` object, specifying a `Time To Live (TTL)` in seconds and an array of `price` objects, each index specifying a `validFrom` filed as an UTCDate and a `honeyPrice`. To summarize, the oracle returns:
+```typescript 
+honeyPrices: { 
+ TTL,
+ prices: [0: {validFrom, honeyPrice}, n: {validFrom, honeyPrice}]
+}
+```
+* The applied price is the `price` where `validFrom` is in the most recent past. 
+* Upon receipt of a payment or intent to send a payment, nodes will check their local cache on the existence of a non-expired `honeyPrice` object. If this exists, the oracle will not be queried, if it does not exist, the oracle will be queried and the answer stored in the local cache of the node for `TTL` time. 
```

Comment:
It would be nice if we could simplify the sentence, "the existence of a non-expired..." and then "exists/not exists" forces to think in negatives. What about
"...will check their local cache for the `honeyPrice` object. If it expired, the oracle will be queried, if not...."

Reply:
Definitely, agree!